pkg/prometheus/agent/statefulset.go

```diff
@@ -155,48 +154,24 @@ func makeStatefulSetSpec(
 ) (*appsv1.StatefulSetSpec, error) {
 	cpf := p.GetCommonPrometheusFields()
 
-	pImagePath, err := operator.BuildImagePath(
-		operator.StringPtrValOrDefault(cpf.Image, ""),
-		operator.StringValOrDefault("", c.PrometheusDefaultBaseImage),
-		operator.StringValOrDefault(cpf.Version, operator.DefaultPrometheusVersion),
-		"",
-		"",
-	)
+	pImagePath, err := buildImagePath(cpf, c)
 	if err != nil {
 		return nil, err
 	}
 
 	cpf.EnableFeatures = append(cpf.EnableFeatures, "agent")
-	promArgs := prompkg.BuildCommonPrometheusArgs(cpf, cg)
-	promArgs = appendAgentArgs(promArgs, cg, cpf.WALCompression)
+	promArgs := buildAgentArgs(cpf, cg)
 
 	var ports []v1.ContainerPort
 	if !cpf.ListenLocal {
-		ports = []v1.ContainerPort{
-			{
-				Name:          cpf.PortName,
-				ContainerPort: 9090,
-				Protocol:      v1.ProtocolTCP,
-			},
-		}
+		ports = prompkg.MakeContainerPorts(cpf)
 	}
 
-	volumes, promVolumeMounts, err := prompkg.BuildCommonVolumes(p, tlsSecrets)
+	volumes, promVolumeMounts, configReloaderVolumeMounts, err := prompkg.BuildVolumes(p, tlsSecrets)
 	if err != nil {
 		return nil, err
 	}
 
-	configReloaderVolumeMounts := []v1.VolumeMount{
```

Comment:
sorry if you answered somewhere else but could this be moved directly to `pkg/operator/config_reloader.go`?

Reply:
Yeah I said in https://github.com/prometheus-operator/prometheus-operator/pull/6686#discussion_r1643751042 that I'd move it to `config_reloader.go`. But then I realized this piece uses `prompkg` fields, so I decided to move it to `prompkg`.

The way I see it, `pkg/operator/config_reloader.go` is just pure config reloader's logic. If `prompkg` fields are used, then it's Prometheus's config reloader container's logic, then it's should be in `prompkg` instead. You agree?

Reply:
:+1: